From ffbd4b502ae30a252fe1a9dfd3267de410b469fc Mon Sep 17 00:00:00 2001
From: Ian Pilcher <arequipeno@gmail.com>
Date: Mon, 7 Aug 2023 09:14:15 +0200
Subject: [PATCH 16/24] leds: trigger: Adapt blkdev_get_by_path() and
 blkdev_put() for Linux 6.5

Sync-up of commits:
  block: introduce holder ops
  block: replace fmode_t with a block-specific type for block open flags
---
 drivers/leds/trigger/ledtrig-blkdev.c | 11 ++++-------
 1 file changed, 4 insertions(+), 7 deletions(-)

diff --git a/drivers/leds/trigger/ledtrig-blkdev.c b/drivers/leds/trigger/ledtrig-blkdev.c
index 067eedb00..9e0c4b66e 100644
--- a/drivers/leds/trigger/ledtrig-blkdev.c
+++ b/drivers/leds/trigger/ledtrig-blkdev.c
@@ -3,7 +3,7 @@
 /*
  *	Block device LED trigger
  *
- *	Copyright 2021-2022 Ian Pilcher <arequipeno@gmail.com>
+ *	Copyright 2021-2023 Ian Pilcher <arequipeno@gmail.com>
  */
 
 #include <linux/blkdev.h>
@@ -47,9 +47,6 @@
  */
 #define BLKDEV_TRIG_CHECK_RETRY	5
 
-/* Mode argument for calls to blkdev_get_by_path() and blkdev_put() */
-#define BLKDEV_TRIG_FMODE	0
-
 /**
  * struct blkdev_trig_bdev - Trigger-specific data about a block device.
  * @last_checked:	Time (in jiffies) at which the trigger last checked this
@@ -539,7 +536,7 @@ static struct block_device *blkdev_trig_get_bdev(const char *path, size_t len)
 	if (buf == NULL)
 		return ERR_PTR(-ENOMEM);
 
-	bdev = blkdev_get_by_path(strim(buf), BLKDEV_TRIG_FMODE, THIS_MODULE);
+	bdev = blkdev_get_by_path(strim(buf), 0, NULL, NULL);
 	kfree(buf);
 	return bdev;
 }
@@ -603,7 +600,7 @@ static struct blkdev_trig_bdev *blkdev_trig_get_btb(const char *path,
 	if (err)
 		devres_free(btb);
 exit_put_bdev:
-	blkdev_put(bdev, BLKDEV_TRIG_FMODE);
+	blkdev_put(bdev, NULL);
 	return err ? ERR_PTR(err) : btb;
 }
 
@@ -786,7 +783,7 @@ static ssize_t unlink_dev_by_path_store(struct device *dev,
 exit_unlock:
 	mutex_unlock(&blkdev_trig_mutex);
 exit_put_bdev:
-	blkdev_put(bdev, BLKDEV_TRIG_FMODE);
+	blkdev_put(bdev, NULL);
 	return err ? : count;
 }
 
-- 
2.42.0

